src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -50,6 +50,34 @@
     </PropertyGroup>
   </Target>
 
+  <!--
+      Target to disable analyzers via an MSBuild property 'RunAnalyzers'. 
+      Note that we remove analyzers only for non-design time builds.
+      Design time builds are background builds inside Visual Studio,
+      see details here: https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md
+      We do not remove analyzers for design time builds to allow users to view the installed analyzers
+      in 'Analyzers' node in the solution explorer and also execute analyzers on-demand with "Run Code Analysis" command.
+      Instead, for design time builds, the VS IDE will read this MSBuild property and skip analyzer execution if it is 'false'.
+  -->
+  <Target Name="_DisableAnalyzers"
+          BeforeTargets="CoreCompile"
+          Condition="'$(RunAnalyzers)' != 'true'">
+    <PropertyGroup>
+      <!-- Determine if this is a design time build: https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md#determining-whether-a-target-is-running-in-a-design-time-build -->
+      <_IsAnyDesignTimeBuild Condition="'$(DesignTimeBuild)' == 'true' or '$(BuildingProject)' != 'true'">true</_IsAnyDesignTimeBuild>
+
+      <_RunAnalyzers>$(RunAnalyzers)</_RunAnalyzers>
+      
+      <!-- Fallback '_RunAnalyzers' to 'RunAnalyzersDuringBuild', the latter property allows users to disable analyzers only for non-design time builds. -->
+      <_RunAnalyzers Condition="'$(_RunAnalyzers)' == ''">$(RunAnalyzersDuringBuild)</_RunAnalyzers>
+    </PropertyGroup>
+
+    <!-- Remove all analyzers for non-design time builds. -->
+    <ItemGroup Condition="'$(_RunAnalyzers)' == 'false' and '$(_IsAnyDesignTimeBuild)' != 'true'">
+      <Analyzer Remove="@(Analyzer)" />
```

Comment:
My Concern with this is the same as #46300 

Because we pass generators in via analyzers, builds can be potentially broken by this now. I think before it was fine: an analyzer can only take a successful build -> failed, but generators are very much the other way.

Reply:
Hmm we already have shipping functionality in VS based on such properties: https://docs.microsoft.com/en-us/visualstudio/code-quality/disable-code-analysis?view=vs-2019. This is done by [this targets file](https://devdiv.visualstudio.com/DevDiv/_git/VS?path=%2Fsrc%2Fedev%2FStaticAnalysis%2Ffxcoptask%2FMicrosoft.CodeAnalysis.targets&version=GBmaster&line=68&lineEnd=85&lineStartColumn=1&lineEndColumn=12&lineStyle=plain) that ships with MSBuild in VS. This change is just ensuring the existing functionality works fine from dotnet builds.

Reply:
Right, I don't think its necessarily an analyzer problem.  

I think there is a separate issue here though, that saying 'Don't run analyzers' is also going to prevent generators from running.

Reply:
Thanks, seems like we can take couple of paths forward here:

1. Abandon this PR and instead have a separate command line compiler flag, say `skipAnalyzers` so the build skips analyzer execution, but not generators.
2. Keep the approach in this PR, and break the `Analyzer` collection into two separate collections `Analyzer` and `Generator`, so this change will not affect generators. However, if user ships both analyzers and generators in the same assembly, then they will be affected.